metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java

```diff
@@ -99,4 +129,182 @@ void testRuleEvaluation() throws URISyntaxException, MalformedURLException {
      */
     metricAnomalyDetector.process(alertTaskBuilder.build());
   }
+
+  @Test
+  void testMetricAnomaly() throws URISyntaxException, MalformedURLException {
+
+    // create mock alertTask
+    LhsExpression lhsExpression =
```

Comment:
abstract out the rule building part

Reply:
menas?
create a function for that ?

Reply:
yes, since the rule building is common in both the tests

Reply:
But the values we are providing may be different in other cases right? So I should pass all the values in the function that we are setting? Like key/scope etc. ?

Reply:
yeah

Reply:
params which do not change between the two test keep it hardcoded